src/Lean/PrettyPrinter/Delaborator/Options.lean

```diff
@@ -125,6 +125,16 @@ register_builtin_option pp.instanceTypes : Bool := {
   group    := "pp"
   descr    := "(pretty printer) when printing explicit applications, show the types of inst-implicit arguments"
 }
+register_builtin_option pp.deepTerms : Bool := {
+  defValue := true
+  group    := "pp"
+  descr    := "(pretty printer) display deeply nested terms"
+}
```

Comment:
Hmm, double negation has its own downsides, and `omit` sounds like a negation, though. `pp.truncateTerms` maybe?

Reply:
I'm not seeing a double negation here -- `pp.omitDeepTerms` being true means that deep terms are omitted. "Deep" doesn't seem like a negative quality to me.

I think that the concept of depth is important here; with `pp.truncateTerms` I wouldn't be able to guess what it's doing, or what it means to truncate a term (I might guess it would mean print it as a string and then use "⋯" if it exceeds some number of characters, which is very different).

Reply:
I think it is more consistent to keep the polarity as is; other pp options like `pp.implicit` -> `pp.explicit` have had their polarities changed so that true = more printing. In lean 3 you would have to set some options true and others false to get `pp.all` behavior a la carte, which was confusing.

I have no opinion regarding the name of the option.

Reply:
One difference between this new option and ones like `pp.explicit` or `pp.proofs` is that turning this one on results in terms that cannot be elaborated. These other options make the pretty printer create terms that are more likely to be successfully elaborated.

Reply:
@kmill I am confused by your statement. Assuming that the polarity is that of `pp.deepTerms`, i.e. false means use `...` and true means to show everything, it would not be any different from `pp.explicit` or `pp.proofs`, more true equals more printing equals more likely to be successfully elaborated. That's exactly my point.

Reply:
@digama0 I'm saying that this feature is different because it has to do with a feature that makes terms impossible to elaborate, rather than just less likely.

It's true though that `pp.proofs` being off makes elaboration vanishingly unlikely.

Reply:
You could use `_` instead of `...` to make it at least syntactically correct, although using a special syntax is also useful for making it clear what has happened.

Reply:
Indeed this discussion makes me wonder whether `pp.proofs` should be reimplemented to use the same UX as elision by depth.

Reply:
> I think it is more consistent to keep the polarity as is; other pp options like `pp.implicit` -> `pp.explicit` have had their polarities changed so that true = more printing

I would agree with that, indeed that was my reasoning behind introducing `pp.explicit`